.github/workflows/ci.yml

```diff
@@ -271,10 +271,14 @@ jobs:
           path: ~/.gradle/wrapper
           key: ${{ runner.os }}-gradle-wrapper-cache-${{ hashFiles('examples/distro/gradle/wrapper/gradle-wrapper.properties') }}
 
-      - name: Local publish
+      - name: Local publish of artifacts
         # javadoc task fails sporadically fetching https://docs.oracle.com/javase/8/docs/api/
         run: ./gradlew publishToMavenLocal -x javadoc
 
+      - name: Local publish of gradle plugins
+        run: ../gradlew publishToMavenLocal -Pversion=1.7.0-alpha-SNAPSHOT
```

Comment:
It looks like the new feature is this publishToMavenLocal, which we don't like due to the version requirement (and not being able to nebula the version).

It may work to do something like this in the examples settings.gradle

```
if (gradle.startParameters["includeGradlePlubins"]) {
  includeBuild("../gradle-plugins") {
  dependencySubstitution {
    substitute(module("io.opentelemetry.instrumentation.muzzle-generation:io.opentelemetry.instrumentation.muzzle-generation.gradle.plugin")).using(project(":"))
    substitute(module("io.opentelemetry.instrumentation.muzzle-check:io.opentelemetry.instrumentation.muzzle-check.gradle.plugin")).using(project(":"))
  }
}
```
a bit similar to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/conventions/settings.gradle.kts#L1 but allowing examples to still build independently too.